src/WorkItemMigrator/JiraExport/JiraProvider.cs

```diff
@@ -393,32 +393,37 @@ public string GetUserEmail(string usernameOrAccountId)
         }
 
         public string GetCustomId(string propertyName)
-        {
+        {   
             var customId = string.Empty;
             if (JiraNameFieldCache == null)
             {
                 var response = (JArray)_jiraServiceWrapper.RestClient.ExecuteRequestAsync(Method.GET, $"{JiraApiV2}/field").Result;
-                JiraNameFieldCache = response
-                    .Where(x => x.Value<string>("name") != null && x.Value<string>("id") != null)
-                    .Select(x => new { name = x.Value<string>("name").ToLower(), id = x.Value<string>("id").ToLower() })
-                    .ToLookup(l => l.name, l => l.id);
-
-                JiraKeyFieldCache = response
-                    .Where(x => x.Value<string>("key") != null && x.Value<string>("id") != null)
-                    .Select(x => new { key = x.Value<string>("key").ToLower(), id = x.Value<string>("id").ToLower() })
-                    .ToLookup(l => l.key, l => l.id);
+                JiraNameFieldCache = CreateFieldCacheLookup(response, "name", "id");
             }
 
             customId = GetItemFromFieldCache(propertyName, JiraNameFieldCache);
 
             if (string.IsNullOrEmpty(customId))
             {
+                if (JiraKeyFieldCache == null)
+                {
+                    var response = (JArray)_jiraServiceWrapper.RestClient.ExecuteRequestAsync(Method.GET, $"{JiraApiV2}/field").Result;
```

Comment:
Add one more change, define the response at the top of the function, then before doing the same API call check if there is any response existing as it might be that in one run you will hit this if-case and you will do the same API call again.

Reply:
Thanks, yes, we can re-use the response this way. We are back to only one fields API call, not two 👍